Feat/gb.im/project

rudqlsdla opened the pull request:
PR 요청합니다!

rudqlsdla committed:
Feat: Splash Screen implementation

- Splash Screen xml coding
- Splash Screen -> Main Screen transition after delay time
- Splash Screen text_loading change animation

rudqlsdla committed:
Feat: Main Screen BottomNavigationBar implementation

- BottomNavigationBar create
- home & bookmark Fragment add

rudqlsdla committed:
Feat: Home Screen RecyclerView frame implementation

- Home Screen RecyclerView xml
- Home Screen RecyclerView's bookmark button logic implementation
- DogDto Create

rudqlsdla committed:
Feat: DogAPI Data GET & DB Implement & Home Screen RecyclerView Implement

- DogAPI Data GET with Retrofit
- DB Implement with Room
- Show Data in RecyclerView

rudqlsdla committed:
Feat: Home Screen RecyclerView Implement

- Show DogImage in RecyclerView with Glide

rudqlsdla committed:
Feat: InformationActivity Implementation

- RecyclerView Click -> InformationActivity
- InformationActivity xml
- Weight & Height Average Calculation
- Ctrl + Shift + L (Code Align)
- Null data -> set '. . . .'

rudqlsdla committed:
Feat: InformationActivity Implementation

- DrawableLeft Image Resizeable And Set

rudqlsdla committed:
Feat: SearchView Implementation

- Operates Clicking Search
- Search Whenever a character Changes

rudqlsdla committed:
Feat: InformationActivity's ToolBar Implementation

- Back Button Click -> Previous Screen

rudqlsdla committed:
Feat: Sync Bookmarks

- SearchBar <-> HomeFragment RecyclerView Sync
- InformationActivity <-> HomeFragment RecyclerView Sync

rudqlsdla committed:
Feat: BookmarkFragment Implementation

- RecyclerView Implementation with GridView

rudqlsdla committed:
Feat: BookmarkFragment Implementation

- Bookmark CRUD Implementation
- DogDao Query Add
- Android's Back Btn Implementation
- Home <-> Bookmark & InformationActivity <-> Bookmark Sync

rudqlsdla committed:
Feat: BookmarkFragment View LongClick Event Implementation

- BookmarkFragment RecyclerView LongClick -> Show AlertDialog
- AlertDialog's Delete Click -> Delete Bookmark
- AlertDialog's Cancel Click -> Cancel AlertDialog

rudqlsdla committed:
Feat: Dog Information Search to Google

- Button Click in InformationActivity -> Search Dog Name to Google

rudqlsdla committed:
Feat: Search Data Implementation

- Search Data is Empty -> No Data Text Appear

rudqlsdla committed:
Feat: Network Check Implementation

- Network Check and DB Insert in SplashActivity
- Bookmark Click in BottomNavigation -> Network Disconnection -> Alert

rudqlsdla committed:
Feat: Infinite Scroll Implementation

- Infinite Scroll Implementation in HomeFragment

rudqlsdla committed:
Refactor : Code Refactoring

wangoose commented on app/src/main/java/com/example/test/activity/SplashActivity.java:
```diff
@@ -0,0 +1,158 @@
+
+        //네트워크 체크 후 로직 실행
+        boolean isConnected = isNetworkConnected(SplashActivity.this);
+        if (!isConnected) { // 인터넷 연결 X
```
isConnected 변수가 한번만 쓰인다면
굳이 멤버 변수를 조건문 위에 생성하지 않고 if문 소괄호 안에서 처리하는게 좋을 것 같아요 😃

wangoose commented on app/src/main/java/com/example/test/activity/SplashActivity.java:
```diff
@@ -0,0 +1,158 @@
+        splashImageView.setImageResource(R.drawable.splash_icon);
+
+        //delayTime 동안 Loading text 변경 애니메이션을 위한 타이머 생성
+        CountDownTimer loadingTimer = new CountDownTimer(DELAY_TIME, 500) { //0.5초마다 onTick 실행
```
CountDownTimer 말고 제가 초반에 설명드렸던
Thread와 delayTime을 활용하여 처리를 하도록 변경하시면 좋을 것 같아요 😢

wangoose commented on app/src/main/java/com/example/test/activity/SplashActivity.java:
```diff
@@ -0,0 +1,158 @@
+            public void onFinish() {
+                Log.d("TAG", "onFinish: 타이머가 끝남");
+                if (db.getDogDao().getAll().isEmpty()) { //DB에 저장된 데이터가 없으면 앱 종료
+                    Toast.makeText(SplashActivity.this, "저장된 데이터가 없으므로 \n네트워크 연결 후 다시 실행해주세요.", Toast.LENGTH_SHORT).show();
```
해당 Toast에서 발생하는 String 구문을
strings.xml 에서 불러와 처리하도록 해보는게 좋을 것 같아요 🧑‍🎓

wangoose commented on app/src/main/java/com/example/test/activity/SplashActivity.java:
```diff
@@ -0,0 +1,158 @@
+    }
+
+    //DB에 받아온 API Data Insert
+    public void insertApiData(DogDto dogDto) {
```
SplashActivity에서만 사용되는 Method이면
Access modifier를 public으로 둘 필요는 없어보입니다 🙂 
또한 메서드명만 보면 "API 데이터를 Insert한다"인데, 어디에 Insert 되는지 모르겠네요  🤔 
메서드명 만으로도 무슨 역할을 하는지 유추가 가능하도록 이름이 길어도 괜찮으니,
명확하게 작명하시는 연습이 필요해보입니다 💇‍♂️

rudqlsdla replied:
insertApiDataIntoDB와 같이 into로 어디에 insert 하는지 표기 해서 메서드명을 작성 해도 괜찮을까요?

wangoose commented on app/src/main/res/layout/activity_splash.xml:
```diff
@@ -0,0 +1,32 @@
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <TextView
+        android:id="@+id/text_loading"
```
이 레이아웃 뿐만 아니라 다른 xml 파일들의 android:id 들도 체크하셔서
누락된 부분이 있다면 모두 camelCase로 변경해주세요 😆

wangoose commented on app/src/main/java/com/example/test/activity/MainActivity.java:
```diff
@@ -0,0 +1,61 @@
+        AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+        builder.setMessage("네트워크 연결이 원활하지 않아 이미지가 로딩되지 않을 수도 있습니다.")
+                .setCancelable(false)
+                .setPositiveButton("CANCEL", new DialogInterface.OnClickListener() {
```
AlertDialog 생성시 기본으로 표시되는 "확인" 대신
"CANCEL"로 표시되는 PositiveButton을 넣으신 이유가 있으신가요? 🤔

rudqlsdla replied:
builder.setMessage(R.string.main_dialog_text).shpw(); 이렇게 코드를 작성하면 
기본으로 표시되는 버튼이 없어서 .setPositiveButton을 추가했어요.

wangoose commented on app/src/main/java/com/example/test/activity/MainActivity.java:
```diff
@@ -0,0 +1,61 @@
+                else if (menuItem.getItemId() == R.id.bookmark) {
+                    //네트워크 연결 안되면 경고창
+                    boolean isConnected = SplashActivity.isNetworkConnected(MainActivity.this);
+                    if (!isConnected) {
```
SplashActivity에서 말씀드렸던 것과 동일합니다!

wangoose commented on app/src/main/java/com/example/test/activity/MainActivity.java:
```diff
@@ -0,0 +1,61 @@
+        getSupportFragmentManager().beginTransaction().add(R.id.main_frame, new HomeFragment()).commit();
+
+        //바텀 네비게이션에서 아이템이 클릭 됐을 때의 리스너를 설정
+        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
```
Deprecated 된(IDE에서 취소선과 함께 Warning) 메서드는 가급적 사용하지 않는 것이 좋습니다.
해결 방법을 찾아서 변경해보는게 좋을 것 같아요. 🙂

wangoose commented on app/src/main/java/com/example/test/activity/SplashActivity.java:
```diff
@@ -0,0 +1,158 @@
+            public void onTick(long millisUntilFinished) {
+                switch (delayCnt) {
+                    case 1:
+                        loadingText.setText("Loading .");
```
Loading Indicator 역할을 하는 . .. ... 앞에 붙는 "Loading" 이 하드코딩 되어있네요.
strings.xml에서 불러오면 좋을 것 같습니다 😄

rudqlsdla committed:
Refactor : strings.xml Refactor

wangoose commented on app/src/main/java/com/example/test/fragment/HomeFragment.java:
```diff
@@ -0,0 +1,280 @@
+        intent.putExtra("id", id);
+        intent.putExtra("imgUrl", imgUrl);
+        intent.putExtra("position", position);
+        startActivityForResult(intent, 1);
```
startActivityForResult, onActivityResult는 Deprecated 되었습니다.
registerForActivityResult 라는 새로운 방법으로 대체 해보는게 좋을 것 같아요 🙂

wangoose reviewed (changes requested):
리뷰한접시

rudqlsdla committed:
Fix : HardCording Fix with strings.xml
Fix : Use the isConnected variable only once

wangoose commented on app/src/main/java/com/example/test/fragment/HomeFragment.java:
```diff
@@ -0,0 +1,280 @@
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
```
onCreate() 내에서 너무 많은 구문을 처리하도록 하지 말고, 기능 단위로 쪼개 메서드화 하는 연습이 필요해 보입니다!

예를 들어 저 경우는 크게 SearchView, RecyclerView 초기화 / No data placeholder 표시여부 확인 으로
행동을 세 종류로 나눌 수 있으니
onCreateView() { initializeSearchView(); initializeRecyclerView(); setNoDataPlaceholderVisibility(); }
와 같이 해당 생명주기 내에서 어떤일을 하는지 한 눈에 알 수 있도록 분류해두는게 보기 좋겠죠? 🤭 

다른 Activity, Fragment의 onCreate 문들도 같이 바꿔보시면 좋을 것 같아요 ✏️

wangoose commented on app/src/main/java/com/example/test/fragment/HomeFragment.java:
```diff
@@ -0,0 +1,280 @@
+
+    private TextView noDataText;
+    private SearchView searchView;
+    private int pageSize = 15;
```
pageNumber와 같이 변동될 수 있는 값은 변수로 선언하는게 맞지만
pageSize와 같이 항상 고정된 값은 상수로 선언하는 것이 좋을 것 같아요 🙂

wangoose commented on app/src/main/java/com/example/test/fragment/HomeFragment.java:
```diff
@@ -0,0 +1,280 @@
+    private SearchView searchView;
+    private int pageSize = 15;
+    private int pageNumber = 1;
+    private int offset = (pageNumber - 1) * pageSize;
```
page offset 값의 생성부를 저런 방식으로 생각하신 것은 좋은 발상입니다! 👍

하지만 Java에서 멤버 변수를 선언하면서 동시에 생성하였으므로, 저 라인은 사실상 offset = 0과 동일합니다. 🤔 

추후 Kotlin으로 변환시, 프로퍼티의 custom getter를 정의하여 리팩토링해보면 좋을 것 같아요.

그냥 offset이라고 하면 무슨 offset인지 모르니, pageOffset등으로 이름 변경도 필요해 보입니다.

wangoose commented on app/src/main/java/com/example/test/fragment/HomeFragment.java:
```diff
@@ -0,0 +1,280 @@
+            homeAdapter.notifyItemInserted(apiDataList.size() - 1);
+
+            Handler handler = new Handler();
+            new Handler().postDelayed(new Runnable() {
```
데이터가 로드되는 즉시 처리해야 하므로, 1000ms delay를 고정으로 사용할 필요는 없을 것 같습니다 😢 
또한 제가 이전에 말씀드렸던 마지막 스크롤시 Loading Indicator가 뜨는 조건,
추가 로드시 리스트 추가와 인디케이터가 삭제 처리되는 로직 순서를 좀 더 개선하면 좋을 것 같아요 🤔

rudqlsdla replied:
delay는 애니메이션 보여주기식으로 넣었는데 지울까요?

wangoose commented on app/src/main/java/adapter/HomeAdapter.java:
```diff
@@ -0,0 +1,149 @@
+    private final onListItemSelectedInterface mListener;
+    private final int VIEW_TYPE_ITEM = 0;
+    private final int VIEW_TYPE_LOADING = 1;
+    private Context mContext;
```
mContext, mListener 와 같은 Hungarian notaion을 배제하고
context, onListItemClickListener 등과 같이 명시적으로 작명하는게 좋을 것 같아요 🙂

wangoose commented on app/src/main/java/adapter/HomeAdapter.java:
```diff
@@ -0,0 +1,149 @@
+    private Context mContext;
+    private ArrayList<DogDto> homeItemList = null;
+
+    public HomeAdapter(Context context, onListItemSelectedInterface listener) {
```
Adapter에서 context를 사용하고자 할 때, 단순히 생성자 파라미터를 통해 사용하는 방법도 있지만
onCreateViewHolder에서 parent ViewGroup를 활용하여 사용하는 방법도 있으니 한번 찾아보세요 😄

wangoose commented on app/src/main/java/adapter/HomeAdapter.java:
```diff
@@ -0,0 +1,149 @@
+
+
+            //북마크 클릭 시 이벤트 처리
+            button.setOnClickListener(view -> {
```
해당 클릭 이벤트 리스너 등록 구문들은 생성자가 아닌
바인딩 처리하는 메서드에 위치하는게 좀 더 자연스러울 것 같습니다 🙂

rudqlsdla committed:
Fix : Change CountDownTimer to Thread

rudqlsdla committed:
Fix : Change android:id to camelCase

rudqlsdla committed:
Fix : Change AlertDialog in MainActivity

rudqlsdla committed:
Fix : NetworkCheck Thread Implementation in MainActivity

rudqlsdla committed:
Fix : Deprecated setOnNavigationItemSelectedListener Resolve

rudqlsdla committed:
Fix : Deprecated startActivityForResult, onActivityResult Resolve

rudqlsdla committed:
1st Feedback

wangoose commented on app/src/main/java/com/example/test/activity/InformationActivity.java:
```diff
@@ -0,0 +1,191 @@
+
+    public void getApiData(){
+        //API로 상세 정보 데이터 GET
+        call = RetrofitClient.getApiService().getSearchData("live_mtPILdV1Vd1b0kcRxjsB1KVMOOHipR18xuUthvy0Y8gH9ZGvNW69RrCip5CErxth", id + "");
```
API 키와 같이 전역적으로 자주 사용되면서 상수인 것은 싱글톤 변수처럼 상수화 하는게 좋아요 🙂  
문제는 현재 이 리포지터리가 public 인데, API 키를 이렇게 누구나 볼 수 있게 되면 무단 도용으로 보안 위험, 요금 폭탄이 됩니다 😱
(구글에 AWS 과금 폭탄 쳐보세요 ㅋㅋ)
build.gradle, BuildConfig 등을 이용해 프로젝트단에서 API 키를 한번 감춰보세요 🤣

wangoose commented on app/src/main/java/com/example/test/activity/InformationActivity.java:
```diff
@@ -0,0 +1,191 @@
+    }
+
+    //강아지 상세 정보 google에 검색해서 띄워주는 기능 구현
+    public void urlSearch(Response<DogDto> response){
```
메서드명은 urlSearch 보다는 searchUrl 처럼 끝이 동사가 아닌 명사로 되는 작명을 하는게 좋습니다 😃

wangoose reviewed (commented):
리뷰DLC

rudqlsdla committed:
Fix : Method Name은 동사->명사로 작성

rudqlsdla committed:
Fix : Fixed Value Variable -> Constant
Fix : Create After Variable Declaration

rudqlsdla committed:
Fix : Use parent.getContext()

rudqlsdla committed:
Fix : Explicit Expression After Exclusion of Hungarian Notation

rudqlsdla committed:
Fix : Hide DOG_API_KEY

1. Add .gradle, /build, local.properties in .gitignore
2. Save Key in local.properties
3. Properties Declaration & Fetch Key & Sync in build.gradle
4. Change the key used in Java Code or Use Hide Key
- 협업할 때는 local.properties 의 내용을 공유하고 전달 받은 팀원은 위 순서를 시행

rudqlsdla committed:
Fix : LoadingIndicator Improvement

rudqlsdla committed:
Fix : Function Unit Method Create

wangoose commented on app/src/main/java/com/example/test/activity/SplashActivity.java:
```diff
@@ -0,0 +1,182 @@
+
+public class SplashActivity extends AppCompatActivity {
+
+    private boolean isRunning = true;
```
리팩토링 진행하신 후 필요없는 import 문이나
더 이상 사용하지 않는 변수, 메서드들은 제거해주는게 좋습니다ㅎㅎ 🙂
현재 IDE에서 확인시, isRunning 이 현재 never used로 warning 처리되어있네요
Alt + 6 단축키나, 커밋 직전에 뜨는 warning을 통해 Problems 들을 한번씩 체킹해주는 습관이 필요합니다 🥰

wangoose commented on app/src/main/java/com/example/test/activity/MainActivity.java:
```diff
@@ -0,0 +1,52 @@
+public class MainActivity extends AppCompatActivity {
+    private boolean isConnected;
+    //바텀 네비게이션 객체 변수
+    private BottomNavigationView bottomNavigationView;
```
해당  bottomNavigationView가 한 메서드에서만 사용되면
굳이 따로 전역 영역으로 빼놓을 필요는 없을 것 같습니다 🙂 
(Field can be converted to a local variable)

wangoose commented on app/src/main/java/com/example/test/fragment/HomeFragment.java:
```diff
@@ -0,0 +1,286 @@
+    public void onItemSelected(View v, int position, ArrayList<DogDto> arrayList) {
+        boolean checkData = false; // 북마크 상태 저장 변수 true -> 북마크 O false -> 북마크 X
+        BookmarkDB bookmarkDB = new BookmarkDB(arrayList, position, ct);
+        checkData = bookmarkDB.dbCheck(); // 데이터가 북마크 되어 있는 상태인지 확인
```
boolean checkData = false; 로 생성한 후
checkData = bookmarkDB.dbCheck();를 하신 이유가 있을까요?
dbCheck() 메서드가 DAO에서 북마크 유무에 따라 boolean return을 하므로
위 문장은 굳이 필요해보이지 않아 보입니다 😢 
(Variable 'checkData' initializer 'false' is redundant)

wangoose commented on app/src/main/java/com/example/test/fragment/BookmarkFragment.java:
```diff
@@ -0,0 +1,125 @@
+    private BookMarkAdapter bookmarkAdapter;
+    private Context ct;
+    private View v;
+    private RecyclerView bookmarkRecyclerView;
```
MainActivity에서 말씀드렸던 것과 같습니다 🙂 
(Field can be converted to a local variable)

wangoose commented on app/src/main/java/com/example/test/fragment/BookmarkFragment.java:
```diff
@@ -0,0 +1,125 @@
+
+    private void initializeTotalTextView(){
+        totalText = v.findViewById(R.id.countText);
+        totalText.setText(bookmarkInfo.size() + " results in bookmark");
```
" results in bookmark" String이 코드 내에 많이 사용되고 있는데
이 문장도 하드코딩 말고 strings.xml 에서 불러오면 좋을 것 같아요 🤣 

또한 해당 String이 사용된 구문들이 공통적으로
아이템이 있다면 텍스트를 바꾸고, 어댑터 notify 작업을 처리하는 동일한 로직이므로
해당 부문도 따로 메서드화 시키면 좋을 것 같습니다 🙂 

이 경우 findViewById 하는 구문은 onCreateView 쪽으로 빼놓으셔도 괜찮을 것 같습니다
(추후 프로젝트에 Data Binding 적용시 findViewById 를 사용하지 않음)

wangoose commented on app/src/main/java/com/example/test/api/RetrofitInterface.java:
```diff
@@ -0,0 +1,29 @@
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+public interface RetrofitInterface {
```
추후 실무에서 앱 제작시 REST API를 기능별로 묶어야 하거나 서로 다른 URL에서 가져오게 되는 경우,
인터페이스를 해당 종류별로 클래스를 만들어 분리 하게 됩니다.
현재는 하나만 사용하지만, 연습삼아 DogAPIService 와 같이 이름을 변경해보세요 🙂

wangoose commented on app/src/main/java/com/example/test/activity/InformationActivity.java:
```diff
@@ -0,0 +1,204 @@
+import retrofit2.Response;
+
+public class InformationActivity extends AppCompatActivity {
+    private Call<DogDto> call;
```
MainActivity에서 말씀드렸던 것과 같습니다 🙂
(Field can be converted to a local variable)

wangoose commented on app/src/main/java/com/example/test/activity/InformationActivity.java:
```diff
@@ -0,0 +1,204 @@
+
+    //강아지 상세 정보 google에 검색해서 띄워주는 기능 구현
+    public void searchUrl(Response<DogDto> response){
+        String url = "https://www.google.com/search?q=" + response.body().getName();
```
Google 검색 URL의 prefix(https://www.google.com/search?q=)가 고정된 String 이므로
상수화 시키는게 좋을 것 같습니다.
참고로 저런 URL은 앱 언어가 달라져도 항상 같으므로 strings.xml 에 저장할 필요는 없습니다 🙂

wangoose commented on app/src/main/java/com/example/test/activity/InformationActivity.java:
```diff
@@ -0,0 +1,204 @@
+        if (response.body().getHeight().getMetric().contains("-")){
+            avgHeight = response.body().getHeight().getHeightAvg().toString();
+        }
+        String result = "avg. " + avgWeight + "kg / " + avgHeight + "cm";
```
avg. kg 표현시 해당 String이 하드코딩이 되어있네요 🤔 
strings.xml 에서 %s, %d 와 같은 String Format을 적용하여 + 연산자로 문자열을 합치지 않고
한줄로 표현해보시겠어요? 🥰 
(https://stackoverflow.com/questions/12627457/format-statement-in-a-string-resource-file)

wangoose commented on app/src/main/java/Interface/onListItemSelectedInterface.java:
```diff
@@ -0,0 +1,13 @@
+
+import java.util.ArrayList;
+
+public interface onListItemSelectedInterface {
```
클래스 뿐만 아니라 인터페이스 명도 PascalCase로 작성하셔야 합니다ㅎㅎ 🤣 
또한 인터페이스 명은 피동 표현 없이 역할에 맞게 OnListItemSelectListener 와 같이 작명하시는게 좋을 것 같습니다 🙂

wangoose commented on app/src/main/java/Interface/onListItemSelectedInterface.java:
```diff
@@ -0,0 +1,13 @@
+import java.util.ArrayList;
+
+public interface onListItemSelectedInterface {
+    void onItemSelected(View v, int position, ArrayList<DogDto> arrayList);
```
이 부분도 쓰이는 맥락에 맞게 onBookmarkSelected, onListItemSelected 등과 같이
어떤 것을 선택했을 때 콜백을 전달하는 추상 메서드인지 좀 더 구체적으로 적는 연습이 필요할 것 같아요 🤭

wangoose commented on app/src/main/java/Interface/onBookmarkListItemSelectedInterface.java:
```diff
@@ -0,0 +1,8 @@
+
+public interface onBookmarkListItemSelectedInterface {
+
+    void changeScreen(int id, String imgUrl, int position);
```
리스너의 추상 메서드에 어떤 작업을 할지 까지는 명시하여 작명하지 않으셔도 됩니다.
그건 implement 한 클래스에서 정하는 거니까요 🤭 

아이템을 선택했을 "때", 아이템을 Long Click을 했을 "때" 등으로
특정 행위가 발생했을 "때"로 추상 메서드 네이밍을 작명하시는 연습이 필요할 것 같습니다 🤔 
(Event handler naming convention 관련하여 한번 검색해보세요)

rudqlsdla committed:
Fix : Refactoring
- Delete never used
- Resolve 'Field can be converted to a local variable'
- Delete Redundant Variable
- Create setBookmarkRecyclerView in BookmarkFragment
- Change API Interface Name
- Change Interface Name, Method Name

wangoose commented on app/src/main/java/com/example/test/activity/SplashActivity.java:
```diff
@@ -0,0 +1,177 @@
+        loadingText = findViewById(R.id.textLoading);
+        ImageView splashImageView = findViewById(R.id.icon);
+        //스플래시 화면에 있는 원형 아이콘 처리
+        splashImageView.setImageResource(R.drawable.splash_icon);
```
ImageView 자체에서 android:src 옵션을 안쓰고 onCreate() 에서
따로 setImageResource 처리를 하신 이유가 있나요? 🤔

rudqlsdla replied:
원형으로 아이콘 처리하려면 저렇게 해야하는 줄 알았는데 xml 에서 처리해도 가능하네요ㅎㅎ..

wangoose commented on app/src/main/java/com/example/test/activity/MainActivity.java:
```diff
@@ -0,0 +1,49 @@
+    }
+
+    private void showNetworkWarningDialog() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
```
AlertDialog를 변수로서 생성 후 show를 할 수도 있지만, 일회성으로 사용되는 변수인 경우

```
new AlertDialog.Builder(context)
    .setMessage()
    .setPosiviteButton()
    .show();
```
와 같이 한 줄로 축약하여 간결한 가독성을 고려하면 좋을 것 같아요 🙂

rudqlsdla committed:
Fix : Refactoring

- Delete never used
- Resolve 'Field can be converted to a local variable'
- Delete Redundant Variable
- Create setBookmarkRecyclerView in BookmarkFragment
- Change API Interface Name
- Change Interface Name, Method Name

rudqlsdla committed:
Merge remote-tracking branch 'origin/feat/gb.im/project' into feat/gb.im/project

wangoose commented on app/src/main/java/Interface/OnBookmarkListItemSelectedInterface.java:
```diff
@@ -0,0 +1,8 @@
+package Interface;
+
+public interface OnBookmarkListItemSelectedInterface {
```
제가 이전에 말씀드렸듯이 Interface 이름 작명시, 네이밍 컨벤션을 참고하여
OnBookmarkListItemSelectListener 와 같이 변경하는게 좋을 것 같습니다 🙂 
인터페이스인 부분은 public 뒤에 붙는 예약어로 확인할 수 있고,
추상 메서드가 아닌 인터페이스 명에는 피동 표현을 제거해야 합니다 🤭

rudqlsdla committed:
Fix : Refactoring

rudqlsdla committed:
Fix : Refactoring
